ros2topic/ros2topic/verb/echo.py

```diff
@@ -82,6 +85,61 @@ def add_arguments(self, parser, cli_name):
         parser.add_argument(
             '--raw', action='store_true', help='Echo the raw binary representation')
 
+    def choose_qos(self, node, args):
+
+        if (args.qos_profile or args.qos_reliability or args.qos_durability or
+            args.qos_depth or args.qos_history):
+                return qos_profile_from_short_keys(args.qos_profile,
+                                               reliability=args.qos_reliability,
+                                               durability=args.qos_durability,
+                                               depth=args.qos_depth,
+                                               history=args.qos_history)
+
+        qos_profile = QoSPresetProfiles.get_from_short_key('sensor_data')
+        reliability_reliable_endpoints_count = 0
+        durability_transient_local_endpoints_count = 0
+        publishers_count = 0
+
+        for info in node.get_publishers_info_by_topic(args.topic_name):
+            publishers_count += 1
+            if (info.qos_profile.reliability ==
+                QoSReliabilityPolicy.RELIABLE):
+                    reliability_reliable_endpoints_count += 1
+            if (info.qos_profile.durability ==
+                QoSDurabilityPolicy.TRANSIENT_LOCAL):
+                    durability_transient_local_endpoints_count += 1
```

Comment:
Minor style comment, feel free to ignore:

---

this is a bit more readable (though it iters twice over the list):

```suggestion
        pubs_info = node.get_publishers_info_by_topic(args.topic_name)
        publishers_count = len(pubs_info)
        reliability_reliable_endpoints_count = sum(
            info.qos_profile.reliability == QoSReliabilityPolicy.RELIABLE for info in pubs_info)
        durability_transient_local_endpoints_count = sum(
            info.qos_profile.durability == QoSDurabilityPolicy.TRANSIENT_LOCAL for info in pubs_info)
```

Reply:
Thanks for pointing this out. I made some changes based on your comments, but not exactly the one you suggested. 

Addressed in: [751f584](https://github.com/gonzodepedro/ros2cli/commit/751f58481e87736d7cf2887b279b900c82b10b85)